docs/storage.conf.5.md

```diff
@@ -41,9 +40,118 @@ The `storage` table supports the following options:
   container storage driver (default is "overlay")
   Default Copy On Write (COW) container storage driver
 
+##STORAGE TABLE OPTIONS
+
 **additionalimagestores**=[]
   Paths to additional congtainer image stores. Usually these are read/only and stored on remote network shares.
 
+## STORAGE TABLE OPTIONS DEVICEMAPPER
+
+  Paths to additional congtainer image stores. Usually these are read/only and stored on remote network shares.
+
+**thinpooldev**=""
+
+     Specifies a custom block storage device to use for the thin pool.
+
+     If using a block device for device mapper storage, it is best to use `lvm` to create and manage the
+     thin-pool volume. This volume is then handed to container runtime to exclusively create snapshot volumes
+     needed for images and containers.
+
+     Managing the thin-pool outside of container runtime daemon makes for the most feature-rich method of
+     having the daemon utilize device mapper thin provisioning as the backing storage for its containers. The
+     highlights of the lvm-based thin-pool management feature include: automatic or interactive thin-pool
+     resize support, dynamically changing thin-pool features, automatic thinp metadata checking when lvm
+     activates the thin-pool, etc.
+
+**basesize**=""
+
+     Specifies the size to use when creating the base device, which limits the size of images and containers.
+     The default value is 10G. Note, thin devices are inherently "sparse", so a 10G device which is mostly
+     empty does not use 10 GB of space on the pool. However, the filesystem will use more space for base
+     images the larger the device is.
+
+     The base device size can be increased at daemon restart which will allow all future images and containers
+     (based on those new images) to be of the new base device size.
+
+     basesize=50G causes the container runtime daemon to increase the base device size to 50G. The container
+     runtime daemon will throw an error if existing base device size is larger than 50G. A user can use this
+     option to expand the base device size however shrinking is not permitted.
+
+     This value affects the system-wide "base" empty filesystem that may already be initialized and inherited
+     by pulled images. Typically, a change to this value requires you to remove and recreate the entire
+     container/storage pool.
+
+**fs**=""
+
+     Specifies the filesystem type to use for the base device. The supported options are `ext4` and `xfs`.
+     The default is `ext4`.
```

Comment:
This should be "xfs" if available (the "xfs" filesystem is built-in to the kernel or its module can be loaded, and "mkfs.xfs" is installed), otherwise "ext4".